src/s_audio_jack.c

```diff
@@ -376,10 +373,10 @@ int jack_open_audio(int inchans, int outchans, t_audiocallback callback)
         /* jack spits out enough messages already, do not warn */
         STUFF->st_inchannels = STUFF->st_outchannels = 0;
         return 1;
-    }
+    } 
     if (status & JackNameNotUnique)
-        jack_client_name(jack_get_client_name(jack_client));
-    logpost(NULL, PD_VERBOSE, "JACK: registered as '%s'", desired_client_name);
+        pd_snprintf(sys_devicename, MAXPDSTRING-1, jack_get_client_name(jack_client));
```

Comment:
hmm, this will overwrite the `sys_devicename` if the name was already taken by JACK.
the consequence is, that ALSA MIDI might or might not use the unique-ified name.

e.g.
1. start `pd -devicename Foo -jack -alsamidi -mididev 1`
   - the JACK client name will be "Foo"
   - the ALSA MIDI client name will be "Foo"
2. start another instance of `pd -devicename Foo -jack -alsamidi -mididev 1`
   - the JACK client name will be "Foo-123"
   - the ALSA MIDI client name will be "Foo"
2. start another instance, this time with `pd -devicename Foo -jack -ossmidi`
   - the JACK client name will be "Foo-124"
   - now select "ALSA-MIDI"  via the Media-menu (and make sure to have some MIDI channels)
   - the ALSA MIDI client name will now be ***"Foo-124"***

personally I find it irritating if the ALSA MIDI name is different depending on *when* i switch to ALSA MIDI.
as mentioned in https://github.com/pure-data/pure-data/pull/2466#issuecomment-2434738595, I think we should not try to store a uniquified JACK client name, and instead always try to use the non-unique name.

Reply:
Fair enough. Pushed another change.

Reply:
One other thing I'm not completely comfortable with - is that if the user uses both` -jackname` and `-devicename`, only the value set by `-jackname` will be used for sys-devicename. Since we are presumably going to deprecate -jackname, I'm not sure that this is something we need to code a fix for, but presumably the fix would be to change the order in which the args are parsed in main. What do you think?

Reply:
??? i cannot reproduce this.

it's the last flag that wins, so if you specify `-jackname Foo -devicename Bar` then the name will be **Bar**.
otoh, if you specify `-devicename Bar -jackname Foo`, the name will be **Foo**.

This is consistent with specifying `-jackname Foo -jackname Bar` (**Bar**) resp `-devicename Bar -devicename Foo` (**Foo**)

Reply:
just to clarify: we loop over the user-provided arguments and check each one if it matches a given string, and then we do something with this.
I *think* you think that instead we are looping over the list of potential arguments, and check whether the currently selected one is present in the list of user-provided arguments. but this is not the case